### PR TITLE
[New Rule] Add detection rules for auth ML jobs

### PR DESCRIPTION
**Pull request by ajosh0504**
This PR adds detection rules for the [Authentication ML job module](https://github.com/elastic/kibana/pull/101840).

### CODE REVIEW

**Comment by ajosh0504** [comment]
@randomuserid You can now add the descriptions and FP info.

**Comment by ajosh0504** [comment]
This is ready for review, @rw-access.

File: rules/ml/ml_auth_rare_source_ip_for_a_user.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+description = """
+A machine learning job detected a user logging in from an IP address that is unusual for the user. This can be due 
+to credentialed access via a compromised account when the user and the threat actor are in different locations. 
+An unusual source IP address for a username could also be due to lateral movement when a compromised account is used 
+to pivot between hosts.
+"""
+false_positives = [
+    """
+    """,
+]
```
Comment (rw-access) [inline rules/ml/ml_auth_rare_source_ip_for_a_user.toml:18]:
```suggestion
```

File: rules/ml/ml_auth_rare_user_logon.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+description = """
+A machine learning job found an unusual user name in the authentication logs. An unusual user name is one way of detecting 
+credentialed access by means of a new or dormant user account. A user account that is normally inactive, because the user 
+has left the organization, which becomes active, may be due to credentialed access using a compromised account password. 
+Threat actors will sometimes also create new users as a means of persisting in a compromised web application.
+"""
+false_positives = [
+    """
+    """,
+]
```
Comment (rw-access) [inline rules/ml/ml_auth_rare_user_logon.toml:18]:
```suggestion
```

File: rules/ml/ml_auth_spike_in_failed_logon_events.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+anomaly_threshold = 75
+author = ["Elastic"]
+description = """
+A machine learning job founs an unusually large spike in authentication failure events. This can be due to password spraying, 
+user enumeration or brute force activity and may be a precursor to account takeover or credentialed access.
+"""
+false_positives = [
+    """
+    """,
+]
```
Comment (rw-access) [inline rules/ml/ml_auth_spike_in_failed_logon_events.toml:16]:
```suggestion
```

File: rules/ml/ml_auth_spike_in_logon_events.toml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+anomaly_threshold = 75
+author = ["Elastic"]
+description = """
+A machine learning job found an unusually large spike in successful authentication events. This can be due to password 
+spraying, user enumeration or brute force activity.
+"""
+false_positives = [
+    """
+    """,
+]
```
Comment (rw-access) [inline rules/ml/ml_auth_spike_in_logon_events.toml:16]:
```suggestion
```

File: rules/ml/ml_auth_spike_in_logon_events_from_a_source_ip.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+anomaly_threshold = 75
+author = ["Elastic"]
+description = """
+A machine learning job found an unusually large spike in successful authentication events events from a particular 
+source IP address. This can be due to password spraying, user enumeration or brute force activity.
+"""
+false_positives = [
+    """
+    """,
+]
```
Comment (rw-access) [inline rules/ml/ml_auth_spike_in_logon_events_from_a_source_ip.toml:16]:
```suggestion
```

File: rules/ml/ml_auth_rare_hour_for_a_user_to_logon.toml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+description = """
+A machine learning job detected a user logging in at a time of day that is unusual for the user.
+This can be due to credentialed access via a compromised account when the user and the threat 
+actor are in different time zones. In addition, unauthorized user activity often takes place 
+during non-business hours.
+"""
+false_positives = [
+    """
+    """,
+]
```
Comment (rw-access) [inline rules/ml/ml_auth_rare_hour_for_a_user_to_logon.toml:18]:
```suggestion
```

**Review by rw-access — COMMENTED** [review]
Should these rules show up in a 7.13 update or only 7.14+?
Were these jobs shipped with 7.13.0, or have they been added since then?

**Comment by ajosh0504** [comment]
> Should these rules show up in a 7.13 update or only 7.14+?
> Were these jobs shipped with 7.13.0, or have they been added since then?
The jobs are going to show up in [7.14](https://github.com/elastic/kibana/pull/101840). Given that the rules should show up in 7.14+?

**Comment by ajosh0504** [comment]
@randomuserid Do you want to add any false positives info?

**Comment by ajosh0504** [comment]
Although, there was this backport [PR](https://github.com/elastic/kibana/pull/102127) that got automatically created and merged, after I merged my main 7.14 PR for these jobs. So should we retain backport for these rules?

**Comment by randomuserid** [comment]
> @randomuserid Do you want to add any false positives info?

yes will add them by tomorrow at latest

**Comment by rw-access** [comment]
> Although, there was this backport PR that got automatically created and merged, after I merged my main 7.14 PR for these jobs. So should we retain backport for these rules?

`main` targets 7.14. If you add `backport: auto`, then it will also show up as an update to `7.13`. Kibana's backport is similar but slightly different because there, `master` means `8.0` and `7.x` means `7.14`

**Comment by ajosh0504** [comment]
Got it. Thanks for clarifying, @rw-access!

File: rules/ml/ml_auth_rare_hour_for_a_user_to_logon.toml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+from = "now-30m"
+interval = "15m"
+license = "Elastic License"
```
Comment (brokensound77) [inline rules/ml/ml_auth_rare_hour_for_a_user_to_logon.toml:21]: All of these `license` should be `v2`

```suggestion
license = "Elastic License v2"
```

**Review by brokensound77 — COMMENTED** [review]
Left a few comments for changes. I noticed none of these have any information in the _investigation guide_ (`note` field), whereas all of the other ML rules do. Not required, just curious.

Also, to prevent another issue of job ID [mismatch](https://github.com/elastic/detection-rules/pull/1287), can you paste the job ID list from Kibana, or add a link to the files so that we can validate the names.

**Comment by ajosh0504** [comment]
@brokensound77 To your note about investigation guide, I am seeing several jobs missing it. Looks like @randomuserid stopped writing those since ~7.9.

**Comment by ajosh0504** [comment]
Also, I'm noticing some other rules are missing v2 in the license field. Should everything be v2? If that's the case, I can create a separate PR updating any rules that are missing it.

**Comment by brokensound77** [comment]
> @brokensound77 The job IDs should match the jobs in [this](https://github.com/elastic/kibana/tree/security_auth_jobs/x-pack/plugins/ml/server/models/data_recognizer/modules/security_auth/ml) Kibana module.

Thanks, the job IDs check out 👍

**Comment by brokensound77** [comment]
> Also, I'm noticing some other rules are missing v2 in the license field. Should everything be v2? If that's the case, I can create a separate PR updating any rules that are missing it.

Good catch, I can update them so that I can add a test for it too

**Comment by randomuserid** [comment]
> @randomuserid You can now add the descriptions and FP info.

done

**Comment by randomuserid** [comment]
> @randomuserid Do you want to add any false positives info?

done

**Comment by ajosh0504** [comment]
@brokensound77 @rw-access Can we get a 👍 on this?

**Review by rw-access — APPROVED** [review]

File: rules/ml/ml_auth_rare_user_logon.toml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+author = ["Elastic"]
+description = """
+A machine learning job found an unusual user name in the authentication logs. An unusual user name is one way of detecting
+credentialed access by means of a new or dormant user account. A user account that is normally inactive, because the user
+has left the organization, which becomes active, may be due to credentialed access using a compromised account password.
+Threat actors will sometimes also create new users as a means of persisting in a compromised web application.
+"""
+false_positives = [
+    """
+    User accounts that are rarely active, such as an SRE or developer logging into a prod server for troubleshooting, may trigger this alert. Under some conditions, a newly created user account may briefly trigger this alert while the model is learning.
```
Comment (brokensound77) [inline rules/ml/ml_auth_rare_user_logon.toml:17]: can you `toml-lint` this

**Review by brokensound77 — APPROVED** [review]
A few of these rule files need `toml-lint` run to fix line breaks, otherwise LGTM 👍 . Nicely done

**Pull request merged**